tests/test_gene_association_file.py

```diff
@@ -0,0 +1,151 @@
+import pandas as pd
+import pytest
+from gene_association_file import process_go_annotations
+
+
+@pytest.fixture
+def temp_csv_files(tmp_path):
+    """Create temporary CSV files for input and output testing."""
+
+    input_path = tmp_path / "annotations_data100.csv"
+    output_path = tmp_path / "normalized_annotation.csv"
+
+    example_csv = """DB,DB_Object_ID,DB_Object_Symbol,Qualifier,GO_ID,DB_Reference,Evidence_Code,With_From,Aspect,DB_Object_Name,Synonym,DB_Object_Type,Taxon,Date,Assigned_By,Annotation_Extension,Gene_Product_Form_ID
+UniProtKB,Q04604,TYR,enables,GO:0005507,GO_REF:0000024,ISS,UniProtKB:Q9ZP19,F,Tyrosinase,TYR|TYRS,protein,taxon:8409,20161118,UniProt"""
+
+    input_path.write_text(example_csv)
+    return input_path, output_path
+
+
+@pytest.fixture
+def evidence_mapping_df():
+    """Fixture providing an example mapping DataFrame for testing."""
+    return pd.DataFrame([
+        {"Evidence_Code": "ISS", "DB_Reference": "GO_REF:0000024", "evidence_type": "ECO:0000250"},
+        {"Evidence_Code": "ND", "DB_Reference": "DEFAULT", "evidence_type": "ECO:0000307"},
+    ])
+
+
+@pytest.fixture
+def annotation_df():
+    """Fixture providing an example annotations DataFrame for testing evidence mapping."""
+    return pd.DataFrame([
+        {"Evidence_Code": "ISS", "publications": "GO_REF:0000024"},
+        {"Evidence_Code": "ND", "publications": "PMID:UNKNOWN"},
+    ])
+
+
+@pytest.mark.unit
+def test_go_annotation_processed(tmp_path):
+    """Test that the processed GO annotation CSV file has correct columns."""
+    output_file = tmp_path / "new_annotations1.csv"
+    process_go_annotations("annotations_data100.csv", str(output_file))
+
+    assert output_file.exists(), "Output file not created."
+    df = pd.read_csv(output_file)
+
+    expected_columns = [
+        "subject", "predicate", "object", "publications", "Evidence_Code", "supporting_objects",
+        "annotation_date", "primary_knowledge_source", "aggregator", "protocol_id", "negated", "evidence_type"
+    ]
+
+    for col in expected_columns:
+        assert col in df.columns, f"Column {col} missing in output."
+
+
+@pytest.mark.integration
+def test_io_behavior(temp_csv_files):
+    """Integration test for I/O behavior of process_go_annotations."""
+    input_path, output_path = temp_csv_files
+
+    process_go_annotations(str(input_path), str(output_path))
```

Comment:
Since you're testing that the output file exists in the next line, you could check that the output file does not exist before you run the code.